### PR TITLE
Make it work with Parity 1.7

### LINKED ISSUE #70: Change default port to 8545 for newer version of Parity
(since dapps and RPC is merged)

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #70

### CODE REVIEW

File: background/web3.js [hunk 1]
```diff
@@ -60,20 +62,38 @@ export default class Web3 {
...
+      redirect: 'error',
+      referrerPolicy: 'origin'
+    };
+    return fetch(`http://${this.DAPPS}/rpc/`, request)
+      .catch(err => {
+        // TODO [ToDr] Get rid of this in the future!
+        // Version 1.7 of Parity runs dapps server on the same port as RPC.
+        // Previous versions were running on :8080.
+        // For backward compatibility we support both cases.
+        const defaultPort = ':' + DEFAULT_CONFIG.DAPPS.split(':')[1];
```
Comment (ngotchac) [inline background/web3.js:82]: Why not using `this.DAPPS` instead of `DEFAULT_CONFIG.DAPPS`?

File: background/web3.js [hunk 2]
```diff
@@ -17,6 +17,8 @@
 import Config, { DEFAULT_CONFIG } from './config';
 import { TRANSPORT_UNINITIALIZED } from '../shared';
 
+const FAILURE = 'Failed to fetch';
```
Comment (ngotchac) [inline background/web3.js]: Should it be a Regex instead of a string comparison ? (eg. not caring about case)

File: background/web3.js [hunk 3]
```diff
@@ -60,20 +62,38 @@ export default class Web3 {
...
+        const newParityDappsPort = ':8545';
+        if (!(err.message === FAILURE && this.DAPPS.endsWith(defaultPort))) {
+          throw err;
+        }
+
+        const newDapps = this.DAPPS.replace(defaultPort, newParityDappsPort);
+        const url = `http://${newDapps}/rpc/`;
+        return fetch(url, request).then(res => {
+          // Update dapps if it succeeds
+          this.DAPPS = newDapps;
```
Comment (ngotchac) [inline background/web3.js:92]: Shouldn't it update the saved config too?

File: background/web3.js (hunk 2 above)
Comment (tomusdrw) [reply background/web3.js]: The error is always like this (it's returned by Chrome this way), I don't really see an advantage of having regex here.

File: background/web3.js (hunk 1 above)
Comment (tomusdrw) [reply background/web3.js:82]: Good point, will update.

File: background/web3.js (hunk 3 above)
Comment (tomusdrw) [reply background/web3.js:92]: I don't really want to overwrite user settings now. Switching between different Parity versions would be a pain in the ass (cause switching to new one would overwrite the config and either old one stops working or we need to do reverse check as well).

We can override config settings when Parity 1.7 get's more stable.

File: background/web3.js (hunk 1 above)
Comment (tomusdrw) [reply background/web3.js:82]: Ah, no sorry. It's a `defaultPort` so we need to take it from `DEFAULT_CONFIG`, few lines later we update current port using `this.DAPPS`

File: background/web3.js (hunk 2 above)
Comment (ngotchac) [reply background/web3.js]: Just looks safer (if an OS adds special characters, if we want to port to Firefox where to message is somehow the same but different casing, wording, etc.)

File: background/web3.js (hunk 3 above)
Comment (ngotchac) [reply background/web3.js:92]: Good point!

File: background/web3.js (hunk 1 above)
Comment (ngotchac) [reply background/web3.js:82]: Ah right, sorry...

**Review by ngotchac — APPROVED** [review]

File: background/web3.js (hunk 2 above)
Comment (tomusdrw) [reply background/web3.js]: Fair enough. Updated!

**Pull request merged**